File: fds-backend/user-management-service/src/main/java/com/yokke/usermanagement/auth/request/ChangePasswordRequest.java
package com.yokke.usermanagement.auth.request;

import com.yokke.usermanagement.auth.validation.NoWhitespace;
import com.yokke.usermanagement.auth.validation.PasswordMatches;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@PasswordMatches

public class ChangePasswordRequest {
    @NoWhitespace
    @Email
    @NotNull
    @NotBlank
    private String oldPassword;
    @NoWhitespace
    @Email
    @NotNull
    @NotBlank
    private String password;
    @NoWhitespace
    @Email
    @NotNull
    @NotBlank
    private String confirmPassword;
}

File: fds-dashboard/src/packages/user-management-service/src/auth/request/change-password.request.ts
import * as z from "zod";

export type ChangePasswordRequest = {
  password: string;
  confirmPassword: string;
  oldPassword: string;
};

export const ChangePasswordSchema = z.object({
  password: z.string(),
  confirmPassword: z.string(),
  oldPassword: z.string(),
});
export type ChangePasswordZod = z.infer<typeof ChangePasswordSchema>;
